Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java

```diff
@@ -1,7 +1,14 @@
 package ktsnvt.tim1.repositories;
 
 import ktsnvt.tim1.model.Event;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface EventRepository extends JpaRepository<Event, Long> {
+    Event findOneByName(String name);
+
+    @Query(value = "select * from event e where lower(e.name) like ?1 and e.category like ?2 and cast(location_id as char) like ?3", nativeQuery = true)
```

Comment:
Query can be converted to method with keywords

Reply:
The case is that if location is null then all events which meet other criteria should be returned. I can not do that with keywords and one method.

Reply:
Convert it to JPQL query instead, use (argument is null or argument = property) for possible null arguments